### DISCUSSION TITLE
Removing Airflow Extras from the Dockerfile

### DISCUSSION THREAD

**Discussion by NadimYounes** [apache/airflow#14055] [General]
Hello all, 

My team and I are attempting to migrate to Airflow (1.10.14) and are looking to adopt the [newer methods](https://airflow.apache.org/docs/apache-airflow/stable/production-deployment.html ) to deploy/maintain our image. We noticed that there are a lot of extras installed by default that we don't need (aws,gcp,slack,azure) and are trying to understand if there is any way we can remove them. I tried using `--build-arg` with a custom constraints file however some of those extras still seem to get installed in the` pip list `by default.

Could anyone provide any advice? 

Thank you

**Comment by potiuk** [comment]
Sure. Use ``AIRFLOW_EXTRAS`` build arg and specify all extras that you need.

http://airflow.apache.org/docs/apache-airflow/stable/production-deployment.html#id1
